installer/PowerToysBootstrapper/bootstrapper/bootstrapper.cpp

```diff
@@ -59,72 +65,102 @@ std::optional<fs::path> extractIcon()
+            MsiEnableLogW(msiSev, msiLogPath.c_str(), INSTALLLOGATTRIBUTES_APPEND);
         }
+        else
+        {
+            logger = spdlog::null_logger_mt("null");
+        }
+        logger->set_pattern("[%L][%d-%m-%C-%T] %v");
+        logger->set_level(severity);
+        spdlog::set_default_logger(std::move(logger));
+        spdlog::set_level(severity);
+        spdlog::flush_every(std::chrono::seconds(5));
+    }
+    catch (...)
+    {
     }
-    return result;
 }
 
 int bootstrapper()
 {
     using namespace localized_strings;
     winrt::init_apartment();
-
-    int nCmdArgs = 0;
-    LPWSTR* argList = CommandLineToArgvW(GetCommandLineW(), &nCmdArgs);
-    const auto cmdArgs = parseCmdArgs(nCmdArgs, argList);
-    std::wostringstream oss;
-    if (cmdArgs.contains(CmdArgs::showHelp))
+    cxxopts::Options options{ "PowerToysBootstrapper" };
+    // clang-format off
+    options.add_options()
+        ("h,help", "Show help")
+        ("no_full_ui", "Use reduced UI for MSI")
+        ("s,silent", "Suppress MSI UI and notifications")
+        ("no_start_pt", "Do not launch PowerToys after the installation is complete")
+        ("skip_dotnet_install", "Skip dotnet 3.X installation even if it's not detected")
+        ("log_level", "Log level. Possible values: off|debug|error", cxxopts::value<std::string>()->default_value("off"));
+    // clang-format on
+    cxxopts::ParseResult cmdArgs;
+    options.allow_unrecognised_options();
+    try
     {
-        oss << "Supported arguments:\n\n";
-        for (auto [arg, _] : knownArgs)
-        {
-            oss << arg << '\n';
-        }
-        MessageBoxW(nullptr, oss.str().c_str(), L"Help", MB_OK | MB_ICONINFORMATION);
+        cmdArgs = options.parse(__argc, const_cast<const char**>(__argv));
+    }
+    catch (...)
+    {
+    }
+    const bool showHelp = cmdArgs["help"].as<bool>();
+    const bool noFullUI = cmdArgs["no_full_ui"].as<bool>();
+    const bool silent = cmdArgs["silent"].as<bool>();
+    const bool skipDotnetInstall = cmdArgs["skip_dotnet_install"].as<bool>();
+    const bool noStartPT = cmdArgs["no_start_pt"].as<bool>();
+    const auto logLevel = cmdArgs["log_level"].as<std::string>();
+    spdlog::level::level_enum severity = spdlog::level::off;
+
+    if (logLevel == "debug")
+    {
+        severity = spdlog::level::debug;
+    }
+    else if (logLevel == "error")
+    {
+        severity = spdlog::level::err;
+    }
+    setup_log(severity);
+    if (showHelp)
+    {
+        std::ostringstream helpMsg;
+        helpMsg << options.help();
+        MessageBoxA(nullptr, helpMsg.str().c_str(), "Help", MB_OK | MB_ICONINFORMATION);
         return 0;
     }
-    if (!cmdArgs.contains(CmdArgs::noFullUI))
+    spdlog::debug("PowerToys Bootstrapper is launched!\nnoFullUI: {}\nsilent: {}\nno_start_pt: {}\nskip_dotnet_install: {}\nlog_level: {}", noFullUI, silent, noStartPT, skipDotnetInstall, logLevel);
+
+    if (!noFullUI)
```

Comment:
If `noFullUI` is true, it seems we run the .msi with the dafault UI level, we should support the actual msiexec arguments and pass them to the .msi.
No need to do that in this PR but we should fix the current arguments to allow maximum flexibility.